src/error.rs

```diff
@@ -54,8 +60,10 @@ impl fmt::Display for DecodeError {
     }
 }
 
+#[cfg(feature = "std")]
```

Comment:
likewise, I think it'd be cleaner to centralize these `cfg`d impl blocks under an inline private module so that the amount of `cfg`s can be reduced.

Reply:
sure

Reply:
done